### PR TITLE
Java ToDo API client

### PR DESCRIPTION
**Pull request by jasonjoh**

### CODE REVIEW

File: sample-api/clients/java/app/src/main/java/todoclient/http/Http.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public static OkHttpClient getHttpClientWithDevCert() {
+        try {
+            final Properties certProperties = new Properties();
+            certProperties.load(Http.class.getResourceAsStream(PROPERTIES_FILE));
+
+            final String certFile = certProperties.getProperty(CERT_FILE_KEY);
+            final String certPassword = certProperties.getProperty(CERT_PASSWORD_KEY);
+            final KeyStore keyStore = KeyStore.getInstance("PKCS12");
+
+            final InputStream pfxFile = Http.class.getResourceAsStream(certFile);
```
Comment (baywet) [inline sample-api/clients/java/app/src/main/java/todoclient/http/Http.java:33]: doesn't the stream need to be closed after loading? (might be chained from the the trust manager factory, ssl context and ok http client if they are all closeable)

File: sample-api/clients/java/app/src/main/java/todoclient/http/Http.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+            keyStore.load(pfxFile, certPassword.toCharArray());
+
+            final TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("X509");
+            trustManagerFactory.init(keyStore);
+
+            final SSLContext sslContext = SSLContext.getInstance("SSL");
+            sslContext.init(null, trustManagerFactory.getTrustManagers(), new SecureRandom());
+
+            return new OkHttpClient.Builder()
```
Comment (baywet) [inline sample-api/clients/java/app/src/main/java/todoclient/http/Http.java:43]: might want to call the kiota lcient facotry instead to get all the default settings from kiota

**Review by baywet — CHANGES_REQUESTED** [review]
thank you so much for putting this together! a couple of comments

File: sample-api/clients/java/app/src/main/java/todoclient/http/Http.java (hunk 2 above)
Comment (jasonjoh) [reply sample-api/clients/java/app/src/main/java/todoclient/http/Http.java]: Great idea, I keep forgetting that we have that.

**Review by baywet — APPROVED** [review]

**Pull request merged**